guides/keras_nlp/quick_tour.py

```diff
@@ -0,0 +1,610 @@
+from tensorflow import keras
+import tensorflow_datasets as tfds
+
+# Use mixed precision for optimal performance
+keras.mixed_precision.set_global_policy("mixed_float16")
+
+"""
+# KerasNLP: Modular NLP Workflows for Keras
+
+
+`keras-nlp` is a natural language processing library that supports users through their
+entire development cycle. Our workflows are built from modular components that have SoTA
+preset weights and architectures when used out-of-the-box and are easily customizable
+when more control is needed.
+
+This library is an extension of the core `keras` API; all high level modules are `Layers`
+or `Models`. If you are familiar with `keras`, congratulations! You already understand
+most of `keras-nlp`.
+
+This guide demonstrates our modular approach using a sentiment analysis example at six
+levels of complexity:
+* Inference with a pretrained classifier
+* Fine tuning a pretrained backbone
+* Fine tuning with user-controlled preprocessing
+* Fine tuning a custom model
+* Pretraining a backbone model
+* Build and train your own transformer from scratch
+
+Throughout our guide we use Professor Keras, the official Keras mascot, as a visual
+reference for the complexity of the material:
+
+![picture](https://drive.google.com/uc?id=1d14Qpmfgjf6zu4z30HBaonH8PYDHgVoU)
+
+
+"""
+
+"""
+# API quickstart
+
+Our highest level API is `keras_nlp.models`. For each `XX` architecture (e.g., `Bert`),
+we offer the following modules:
+* **Tokenizer**: `keras_nlp.models.XXTokenizer`
+    * Maps raw text to `tf.RaggedTensor`s of token ids.
+    * Inherits from `keras.Layer`.
+* **Preprocessor**: `keras_nlp.models.XXPreprocessor`
+    * Maps raw text to a dictonary of dense tensors consumed by the model.
+    * Has a `XXTokenizer`.
+    * Inherits from `keras.Layer`.
+* **Backbone**: `keras_nlp.models.XXBackbone`
+    * Maps preprocessed tensors to dense representation. *Does not handle raw text*.
+    * Inherits from `keras.Model`.
+* **Task**: e.g., `keras_nlp.models.XXClassifier`
+    * Maps raw text to task-specific output (e.g., classification probabilities).
+    * Has a `XXBackbone` and `XXPreprocessor`.
+    * Inherits from `keras.Model`.
+
+Here is the modular hierarchy for `BertClassifier` (all relationships are compositional):
+
+![picture](https://drive.google.com/uc?id=1vHBQ1oFbto8ItfhsLcxKhIwOIdJE1X9n)
+
+All modules can be used independently and have a `from_preset()` method in addition to
+the standard constructor that instantiates the class with **preset** architecture and
+weights (see examples below).
+"""
+
+"""
+# Data
+
+We will use a running example of sentiment analysis of IMDB movie reviews. In this task,
+we use the text to predict whether the review was positive (`label = 1`) or negative
+(`label = 0`).
+
+We load the data from `tensorflow_datasets`, a collection of machine learning benchmarks
+that uses the powerful `tf.data.Dataset` format for examples.
+"""
+
+BATCH_SIZE = 16
+imdb_train, imdb_test = tfds.load(
```

Comment:
I'd encourage you to start from raw text files rather than TFDS, because it is closer to the kind of workflows that users go through. It only takes a few lines, and these few lines will better generalize to users' own datasets. https://keras.io/examples/nlp/text_classification_from_scratch/#load-the-data-imdb-movie-review-sentiment-classification